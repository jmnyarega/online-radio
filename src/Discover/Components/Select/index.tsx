import React from "react";

type Iprops = {
  getCountry?: (x: any) => void;
  country: string;
  countries: Icountry[];
};

type Icountry = {
  name: string;
};

const Select = ({ getCountry, country, countries }: Iprops) => (
  <span className="search-custom__select row">
    <select
      className="search__select-country block"
      onChange={getCountry}
      value={country}
    >
      <option value="">---Select Country---</option>
      {countries
        .filter((x) => !x.name.startsWith("http"))
        .map((country, i) => {
          return (
            <option
              key={i}
              value={country.name}
              className="search__select-data"
            >
              {country.name}
            </option>
          );
        })}
    </select>
    <span className="search-custom__arrow arrow"></span>
  </span>
);

export default Select;
