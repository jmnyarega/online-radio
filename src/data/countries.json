[
  { "name": ".", "stationcount": 1 },
  { "name": "100e7", "stationcount": 2 },
  { "name": "a", "stationcount": 1 },
  { "name": "Afghanistan", "stationcount": 111 },
  { "name": "Albania", "stationcount": 23 },
  { "name": "Algeria", "stationcount": 78 },
  { "name": "American Samoa", "stationcount": 1 },
  { "name": "Andorra", "stationcount": 5 },
  { "name": "Angola", "stationcount": 11 },
  { "name": "Anguilla", "stationcount": 2 },
  { "name": "Antigua & Barbuda", "stationcount": 3 },
  { "name": "Argentina", "stationcount": 292 },
  { "name": "arica chile", "stationcount": 1 },
  { "name": "Armenia", "stationcount": 27 },
  { "name": "Array", "stationcount": 2 },
  { "name": "Aruba", "stationcount": 10 },
  { "name": "ASD", "stationcount": 1 },
  { "name": "Atlanta ,US", "stationcount": 1 },
  { "name": "Australia", "stationcount": 290 },
  { "name": "Austria", "stationcount": 215 },
  { "name": "Azerbaijan", "stationcount": 10 },
  { "name": "Bahamas", "stationcount": 8 },
  { "name": "Bahrain", "stationcount": 1 },
  { "name": "Bangladesh", "stationcount": 21 },
  { "name": "Barbados", "stationcount": 8 },
  { "name": "Belarus", "stationcount": 60 },
  { "name": "Belgique", "stationcount": 1 },
  { "name": "Belgium", "stationcount": 254 },
  { "name": "Bermuda", "stationcount": 6 },
  { "name": "BG", "stationcount": 1 },
  { "name": "Bolivia", "stationcount": 14 },
  { "name": "Bolivia (Plurinational State of)", "stationcount": 5 },
  { "name": "Bolivia, Plurinational State of", "stationcount": 1 },
  { "name": "Bonaire", "stationcount": 5 },
  { "name": "Bosnia and Herzegovina", "stationcount": 37 },
  { "name": "Botswana", "stationcount": 2 },
  { "name": "Brasil", "stationcount": 23 },
  { "name": "Brazil", "stationcount": 581 },
  { "name": "Brazil - RS - Canoas", "stationcount": 1 },
  { "name": "britain", "stationcount": 2 },
  { "name": "British Indian Ocean Territory", "stationcount": 1 },
  { "name": "Brunei", "stationcount": 1 },
  { "name": "Brunei Darussalam", "stationcount": 2 },
  { "name": "Bulgaria", "stationcount": 115 },
  { "name": "Burkina Faso", "stationcount": 3 },
  { "name": "Burundi", "stationcount": 1 },
  { "name": "Cabada", "stationcount": 1 },
  { "name": "Cambodia", "stationcount": 9 },
  { "name": "Canada", "stationcount": 906 },
  { "name": "Cape Verde", "stationcount": 11 },
  { "name": "Cayman Islands", "stationcount": 4 },
  { "name": "Central African Republic", "stationcount": 1 },
  { "name": "Charleston", "stationcount": 1 },
  { "name": "Chile", "stationcount": 143 },
  { "name": "China", "stationcount": 1400 },
  { "name": "Col", "stationcount": 1 },
  { "name": "cologne", "stationcount": 1 },
  { "name": "Colombia", "stationcount": 203 },
  { "name": "Comoros", "stationcount": 1 },
  { "name": "Congo (Democratic Republic of the)", "stationcount": 1 },
  { "name": "Cook Islands", "stationcount": 1 },
  { "name": "Costa Rica", "stationcount": 41 },
  { "name": "Côte d'Ivoire", "stationcount": 1 },
  { "name": "ČR", "stationcount": 3 },
  { "name": "Croatia", "stationcount": 123 },
  { "name": "Cuba", "stationcount": 47 },
  { "name": "Curaçao", "stationcount": 6 },
  { "name": "Cyprus", "stationcount": 35 },
  { "name": "Czech Republic", "stationcount": 209 },
  { "name": "Democratic Republic of the Congo", "stationcount": 6 },
  { "name": "Denmark", "stationcount": 81 },
  { "name": "Deutschland", "stationcount": 18 },
  { "name": "Dominica", "stationcount": 2 },
  { "name": "Dominican Republic", "stationcount": 33 },
  { "name": "Ecuador", "stationcount": 149 },
  { "name": "Egypt", "stationcount": 27 },
  { "name": "El Salvador", "stationcount": 11 },
  { "name": "Elite: Dangerous", "stationcount": 1 },
  { "name": "England", "stationcount": 8 },
  { "name": "España", "stationcount": 10 },
  { "name": "Estados Unidos", "stationcount": 5 },
  { "name": "Estonia", "stationcount": 55 },
  { "name": "Falkland Islands (Malvinas)", "stationcount": 2 },
  { "name": "Faroe Islands", "stationcount": 6 },
  { "name": "Fiji", "stationcount": 7 },
  { "name": "Finland", "stationcount": 119 },
  { "name": "France", "stationcount": 1761 },
  { "name": "France 🇫🇷", "stationcount": 1 },
  { "name": "french", "stationcount": 1 },
  { "name": "French Guiana", "stationcount": 1 },
  { "name": "Gabon", "stationcount": 1 },
  { "name": "Gambia", "stationcount": 1 },
  { "name": "Georgia", "stationcount": 23 },
  { "name": "German", "stationcount": 1 },
  { "name": "Germany", "stationcount": 2890 },
  { "name": "Germany,hessen", "stationcount": 6 },
  { "name": "Ghana", "stationcount": 7 },
  { "name": "Gibraltar", "stationcount": 2 },
  { "name": "girona", "stationcount": 1 },
  { "name": "Greece", "stationcount": 320 },
  { "name": "Greenland", "stationcount": 3 },
  { "name": "Grenada", "stationcount": 6 },
  { "name": "Guadeloupe", "stationcount": 7 },
  { "name": "Guam", "stationcount": 3 },
  { "name": "Guatemala", "stationcount": 29 },
  { "name": "Guinea", "stationcount": 6 },
  { "name": "Guyana", "stationcount": 3 },
  { "name": "Haiti", "stationcount": 9 },
  { "name": "Holland", "stationcount": 3 },
  { "name": "Honduras", "stationcount": 23 },
  { "name": "Hong Kong", "stationcount": 14 },
  { "name": "http://192.99.1.139:8982/", "stationcount": 1 },
  { "name": "http://mcrscast1.mcr.iol.pt/comercial.mp3", "stationcount": 1 },
  { "name": "http://stream.laut.fm/truckersmp", "stationcount": 1 },
  {
    "name": "http://stream.simulatorradio.com:8002/stream.mp3",
    "stationcount": 1
  },
  {
    "name": "http://stream01.ilovemusic.de/iloveradio20.mp3",
    "stationcount": 1
  },
  { "name": "http://www.radiofaaz.com/radiofaaz", "stationcount": 1 },
  { "name": "https://live.radio.si/Radio1", "stationcount": 1 },
  { "name": "https://onlineradiobox.com/sj/arcticoutpost/", "stationcount": 1 },
  { "name": "https://rj1.rjstream.com/", "stationcount": 1 },
  { "name": "https://simulatorradio.com/", "stationcount": 1 },
  { "name": "https://stream.open.fm/64", "stationcount": 1 },
  { "name": "https://stream.playradio.rs:8443/play.mp3", "stationcount": 1 },
  {
    "name": "https://streams.ilovemusic.de/iloveradio1.mp3",
    "stationcount": 1
  },
  {
    "name": "https://streams.ilovemusic.de/iloveradio104.mp3",
    "stationcount": 1
  },
  { "name": "https://www.bayern3.de", "stationcount": 1 },
  { "name": "https://www.capitalfm.com", "stationcount": 1 },
  { "name": "https://www.youtube.com/watch?v=5qap5aO4i9A", "stationcount": 1 },
  { "name": "https://www.youtube.com/watch?v=UEK0ipXk21k", "stationcount": 1 },
  { "name": "https://youtu.be/5qap5aO4i9A", "stationcount": 1 },
  { "name": "Hungary", "stationcount": 217 },
  { "name": "Ialia", "stationcount": 1 },
  { "name": "Iceland", "stationcount": 29 },
  { "name": "iLoveRadio", "stationcount": 1 },
  { "name": "India", "stationcount": 260 },
  { "name": "Indonesia", "stationcount": 109 },
  { "name": "Internet", "stationcount": 6 },
  { "name": "Iran", "stationcount": 16 },
  { "name": "Iran (Islamic Republic of)", "stationcount": 9 },
  { "name": "Iran, Islamic Republic of", "stationcount": 3 },
  { "name": "Iraq", "stationcount": 5 },
  { "name": "Ireland", "stationcount": 102 },
  { "name": "Isle of Man", "stationcount": 5 },
  { "name": "Israel", "stationcount": 138 },
  { "name": "Italia", "stationcount": 6 },
  { "name": "Italy", "stationcount": 1121 },
  { "name": "Ivory Coast", "stationcount": 6 },
  { "name": "Jamaica", "stationcount": 14 },
  { "name": "Japan", "stationcount": 94 },
  { "name": "jinan", "stationcount": 1 },
  { "name": "Jordan", "stationcount": 8 },
  { "name": "Kazakhstan", "stationcount": 32 },
  { "name": "Kenya", "stationcount": 19 },
  { "name": "Korea (Democratic People's Republic of)", "stationcount": 1 },
  { "name": "Korea (Republic of)", "stationcount": 6 },
  { "name": "Kosovo", "stationcount": 9 },
  { "name": "Kuwait", "stationcount": 3 },
  { "name": "Kyrgyzstan", "stationcount": 2 },
  { "name": "Latvia", "stationcount": 58 },
  { "name": "Lebanon", "stationcount": 21 },
  { "name": "Lëtzebuerg", "stationcount": 1 },
  { "name": "Libya", "stationcount": 10 },
  { "name": "Liechtenstein", "stationcount": 5 },
  { "name": "Lithuania", "stationcount": 64 },
  { "name": "Los Angeles , California", "stationcount": 1 },
  { "name": "lul", "stationcount": 1 },
  { "name": "Luxembourg", "stationcount": 23 },
  { "name": "Macau", "stationcount": 1 },
  { "name": "Macedonia (the former Yugoslav Republic of)", "stationcount": 4 },
  { "name": "Madagascar", "stationcount": 7 },
  { "name": "Malawi", "stationcount": 1 },
  { "name": "Malaysia", "stationcount": 17 },
  { "name": "Maldives", "stationcount": 1 },
  { "name": "Mali", "stationcount": 5 },
  { "name": "Malta", "stationcount": 29 },
  { "name": "Manchester, UK", "stationcount": 1 },
  { "name": "Martinique", "stationcount": 4 },
  { "name": "Mauritius", "stationcount": 2 },
  { "name": "Mayotte", "stationcount": 1 },
  { "name": "Mexico", "stationcount": 213 },
  { "name": "Moldova", "stationcount": 36 },
  { "name": "Moldova (Republic of)", "stationcount": 6 },
  { "name": "Monaco", "stationcount": 5 },
  { "name": "Mongolia", "stationcount": 6 },
  { "name": "Montenegro", "stationcount": 8 },
  { "name": "Monterey, CA", "stationcount": 1 },
  { "name": "Morocco", "stationcount": 70 },
  { "name": "Morroco", "stationcount": 2 },
  { "name": "Mozambique", "stationcount": 6 },
  { "name": "Myanmar", "stationcount": 9 },
  { "name": "Namibia", "stationcount": 7 },
  { "name": "nederland", "stationcount": 3 },
  { "name": "Nepal", "stationcount": 11 },
  { "name": "Netherlands", "stationcount": 452 },
  { "name": "New Caledonia", "stationcount": 3 },
  { "name": "New Zealand", "stationcount": 59 },
  { "name": "Nicaragua", "stationcount": 8 },
  { "name": "Nigeria", "stationcount": 18 },
  { "name": "NJ", "stationcount": 1 },
  { "name": "NL", "stationcount": 1 },
  { "name": "Norge", "stationcount": 3 },
  { "name": "North Korea", "stationcount": 1 },
  { "name": "North Macedonia", "stationcount": 4 },
  { "name": "Norway", "stationcount": 100 },
  { "name": "nrj", "stationcount": 1 },
  { "name": "NRJ France", "stationcount": 1 },
  { "name": "o", "stationcount": 2 },
  { "name": "Ohio", "stationcount": 1 },
  { "name": "Oman", "stationcount": 1 },
  { "name": "online", "stationcount": 1 },
  { "name": "Österreich", "stationcount": 9 },
  { "name": "Pakistan", "stationcount": 14 },
  { "name": "Palestine", "stationcount": 9 },
  { "name": "Panama", "stationcount": 11 },
  { "name": "Papua New Guinea", "stationcount": 3 },
  { "name": "Paraguay", "stationcount": 19 },
  { "name": "Peru", "stationcount": 199 },
  { "name": "Philippines", "stationcount": 33 },
  { "name": "playing", "stationcount": 13 },
  { "name": "Poland", "stationcount": 748 },
  { "name": "Polska", "stationcount": 8 },
  { "name": "Portugal", "stationcount": 231 },
  { "name": "Puerto Rico", "stationcount": 20 },
  { "name": "Qatar", "stationcount": 5 },
  { "name": "Republic of Kosovo", "stationcount": 3 },
  { "name": "Republic of Panama", "stationcount": 1 },
  { "name": "Republic of the Congo", "stationcount": 2 },
  { "name": "República Dominicana", "stationcount": 1 },
  { "name": "resgate", "stationcount": 1 },
  { "name": "Réunion", "stationcount": 5 },
  { "name": "Romania", "stationcount": 201 },
  { "name": "Russia\n", "stationcount": 15 },
  { "name": "Russia", "stationcount": 765 },
  { "name": "Russian Federation", "stationcount": 243 },
  { "name": "Rwanda", "stationcount": 3 },
  { "name": "s", "stationcount": 1 },
  { "name": "Saint Kitts and Nevis", "stationcount": 1 },
  { "name": "Saint Lucia", "stationcount": 7 },
  { "name": "Saint Pierre and Miquelon", "stationcount": 1 },
  { "name": "Saint Vincent and the Grenadines", "stationcount": 6 },
  { "name": "San Francisco, CA, US", "stationcount": 1 },
  { "name": "San Jose California", "stationcount": 1 },
  { "name": "San Marino", "stationcount": 6 },
  { "name": "Saudi Arabia", "stationcount": 16 },
  { "name": "Scotland", "stationcount": 2 },
  { "name": "Scottland", "stationcount": 1 },
  { "name": "Senegal", "stationcount": 6 },
  { "name": "Serbia", "stationcount": 120 },
  { "name": "Seychelles", "stationcount": 1 },
  { "name": "Sierra Leone", "stationcount": 4 },
  { "name": "Singapore", "stationcount": 26 },
  { "name": "Slovakia", "stationcount": 83 },
  { "name": "Slovenia", "stationcount": 58 },
  { "name": "Slovensko", "stationcount": 1 },
  { "name": "South Africa", "stationcount": 52 },
  { "name": "South Florida", "stationcount": 1 },
  { "name": "South Korea", "stationcount": 51 },
  { "name": "South Sudan", "stationcount": 1 },
  { "name": "Space", "stationcount": 1 },
  { "name": "Spain", "stationcount": 589 },
  { "name": "Sri Lanka", "stationcount": 30 },
  { "name": "St. Helena", "stationcount": 1 },
  { "name": "Stuttgart", "stationcount": 1 },
  { "name": "Sudan", "stationcount": 3 },
  { "name": "Suisse", "stationcount": 4 },
  { "name": "Suriname", "stationcount": 6 },
  { "name": "Sverige", "stationcount": 2 },
  { "name": "Sweden", "stationcount": 108 },
  { "name": "Swiss", "stationcount": 1 },
  { "name": "Switzerland", "stationcount": 469 },
  { "name": "Syria", "stationcount": 11 },
  { "name": "Syrian Arab Republic", "stationcount": 1 },
  { "name": "Taiwan", "stationcount": 33 },
  { "name": "Tajikistan", "stationcount": 2 },
  { "name": "Tanzania", "stationcount": 4 },
  { "name": "Tanzania, United Republic of", "stationcount": 1 },
  { "name": "Thailand", "stationcount": 68 },
  { "name": "The Netherlands", "stationcount": 5 },
  { "name": "the us", "stationcount": 1 },
  { "name": "Tibet", "stationcount": 1 },
  { "name": "Togo", "stationcount": 1 },
  { "name": "Tonga", "stationcount": 1 },
  { "name": "Transamerica", "stationcount": 1 },
  { "name": "Trinidad and Tobago", "stationcount": 16 },
  { "name": "Tunisia", "stationcount": 28 },
  { "name": "Turkey", "stationcount": 189 },
  { "name": "Turkmenistan", "stationcount": 1 },
  { "name": "Turks & Caicos Islands", "stationcount": 1 },
  { "name": "Uganda", "stationcount": 8 },
  { "name": "UK", "stationcount": 20 },
  { "name": "Ukraine", "stationcount": 128 },
  { "name": "United Arab Emirates", "stationcount": 18 },
  { "name": "United Kingdom", "stationcount": 560 },
  {
    "name": "United Kingdom of Great Britain and Northern Irela",
    "stationcount": 102
  },
  { "name": "United Kingdowm", "stationcount": 1 },
  { "name": "United States", "stationcount": 15 },
  { "name": "United States Minor Outlying Islands", "stationcount": 1 },
  { "name": "United States of America", "stationcount": 3262 },
  { "name": "Uruguay", "stationcount": 42 },
  { "name": "US", "stationcount": 1 },
  { "name": "US Virgin Islands", "stationcount": 13 },
  { "name": "USA", "stationcount": 1663 },
  { "name": "Uzbekistan", "stationcount": 4 },
  { "name": "Vanuatu", "stationcount": 1 },
  { "name": "Vatican City State", "stationcount": 12 },
  { "name": "Venezuela", "stationcount": 10 },
  { "name": "Venezuela (Bolivarian Republic of)", "stationcount": 6 },
  { "name": "Venezuela, Bolivarian Republic of", "stationcount": 1 },
  { "name": "Viet Nam", "stationcount": 4 },
  { "name": "Vietnam", "stationcount": 33 },
  { "name": "VN", "stationcount": 1 },
  { "name": "Warsaw", "stationcount": 1 },
  { "name": "Warszawa", "stationcount": 1 },
  { "name": "Winston-Salem, North Carolina, USA", "stationcount": 1 },
  { "name": "world", "stationcount": 1 },
  { "name": "Yemen", "stationcount": 2 },
  { "name": "youtube.com", "stationcount": 1 },
  { "name": "Zambia", "stationcount": 3 },
  { "name": "Zimbabwe", "stationcount": 7 },
  { "name": "Россия", "stationcount": 5 },
  { "name": "山东", "stationcount": 1 },
  { "name": "🎤nrj🎤", "stationcount": 1 }
]
