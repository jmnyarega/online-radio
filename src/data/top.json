[
  {
    "changeuuid": "6fdd6fce-1148-11ea-a87e-52543be04c81",
    "stationuuid": "6fdd6fbf-1148-11ea-a87e-52543be04c81",
    "name": "Prambors otopulsa.com",
    "url": "http://masima.rastream.com/masima-pramborsjakarta?awparams=stationid:masima-pramborsjakarta",
    "url_resolved": "http://masima.rastream.com/masima-pramborsjakarta?awparams=stationid:masima-pramborsjakarta",
    "homepage": "http://otopulsa.com",
    "favicon": "https://aksesglobalmultimedia.com/radio/upload/prambors.jpg",
    "tags": "pop,indonesia,oto pulsa,radio online agm,otopulsa",
    "country": "Indonesia",
    "countrycode": "ID",
    "state": "Jawa Barat",
    "language": "indonesia",
    "votes": 1579412,
    "lastchangetime": "2020-01-23 20:40:50",
    "codec": "AAC",
    "bitrate": 64,
    "hls": 0,
    "lastcheckok": 1,
    "lastchecktime": "2020-09-04 09:23:45",
    "lastcheckoktime": "2020-09-04 09:23:45",
    "lastlocalchecktime": "2020-09-04 09:23:45",
    "clicktimestamp": "2020-09-04 00:57:19",
    "clickcount": 87,
    "clicktrend": 0
  },
  {
    "changeuuid": "78012218-1aa1-11e9-a80b-52543be04c81",
    "stationuuid": "78012206-1aa1-11e9-a80b-52543be04c81",
    "name": "MANGORADIO",
    "url": "http://stream.mangoradio.de",
    "url_resolved": "http://mangoradio.stream.laut.fm/mangoradio?t302=2020-09-04_04-33-31&uuid=49b0c809-28d8-40b2-b9f2-3681e78d7d24",
    "homepage": "https://mangoradio.de",
    "favicon": "https://mangoradio.de//android-icon-192x192.png",
    "tags": "webradio,public radio,mongoradio,mangoradio,mongo,mango",
    "country": "Germany",
    "countrycode": "DE",
    "state": "",
    "language": "german",
    "votes": 469707,
    "lastchangetime": "2020-01-23 20:40:50",
    "codec": "MP3",
    "bitrate": 128,
    "hls": 0,
    "lastcheckok": 1,
    "lastchecktime": "2020-09-04 02:40:13",
    "lastcheckoktime": "2020-09-04 02:40:13",
    "lastlocalchecktime": "2020-09-03 17:55:23",
    "clicktimestamp": "2020-09-03 18:56:13",
    "clickcount": 87,
    "clicktrend": 0
  },
  {
    "changeuuid": "f592bd17-c052-11e9-8502-52543be04c81",
    "stationuuid": "f592bcd7-c052-11e9-8502-52543be04c81",
    "name": "REYFM - #original",
    "url": "https://listen.reyfm.de/original_192kbps.mp3",
    "url_resolved": "https://reyfm-stream04.radiohost.de/reyfm-original_mp3-192",
    "homepage": "https://www.reyfm.de/",
    "favicon": "https://www.reyfm.de/img/fav.png",
    "tags": "#original,fm,rey,reyfm",
    "country": "",
    "countrycode": "",
    "state": "",
    "language": "",
    "votes": 149052,
    "lastchangetime": "2020-01-23 20:40:50",
    "codec": "MP3",
    "bitrate": 192,
    "hls": 0,
    "lastcheckok": 1,
    "lastchecktime": "2020-09-04 07:57:17",
    "lastcheckoktime": "2020-09-04 07:57:17",
    "lastlocalchecktime": "2020-09-04 07:57:17",
    "clicktimestamp": "2020-09-04 04:31:57",
    "clickcount": 81,
    "clicktrend": 0
  },
  {
    "changeuuid": "a2bd6961-077e-11e8-ae97-52543be04c81",
    "stationuuid": "962cc6df-0601-11e8-ae97-52543be04c81",
    "name": "Dance Wave!",
    "url": "http://dancewave.online/dance.mp3.pls",
    "url_resolved": "http://stream.dancewave.online:8080/dance.mp3",
    "homepage": "http://dancewave.online/",
    "favicon": "http://dancewave.online/dw-logo_s.png",
    "tags": "house,trance,dance",
    "country": "Hungary",
    "countrycode": "HU",
    "state": "",
    "language": "english",
    "votes": 108989,
    "lastchangetime": "2020-01-23 20:40:49",
    "codec": "MP3",
    "bitrate": 128,
    "hls": 0,
    "lastcheckok": 1,
    "lastchecktime": "2020-09-04 07:30:54",
    "lastcheckoktime": "2020-09-04 07:30:54",
    "lastlocalchecktime": "2020-09-04 02:50:36",
    "clicktimestamp": "2020-09-04 08:36:55",
    "clickcount": 88,
    "clicktrend": 0
  },
  {
    "changeuuid": "70f691d6-7d88-11e9-aa30-52543be04c81",
    "stationuuid": "70f691c6-7d88-11e9-aa30-52543be04c81",
    "name": "Alles techno",
    "url": "http://techno.stream.alles-radio-web.de",
    "url_resolved": "http://alles-techno.stream.laut.fm/alles-techno",
    "homepage": "https://alles-radio-web.de",
    "favicon": "https://alles-radio-web.de/assets/img/icon.png",
    "tags": "",
    "country": "Germany",
    "countrycode": "DE",
    "state": "North Rhine-Westphalia",
    "language": "deutsch",
    "votes": 29116,
    "lastchangetime": "2020-01-23 20:40:50",
    "codec": "MP3",
    "bitrate": 128,
    "hls": 0,
    "lastcheckok": 1,
    "lastchecktime": "2020-09-04 08:41:43",
    "lastcheckoktime": "2020-09-04 08:41:43",
    "lastlocalchecktime": "2020-09-03 19:26:49",
    "clicktimestamp": "2020-09-04 04:46:41",
    "clickcount": 84,
    "clicktrend": 0
  }
]
