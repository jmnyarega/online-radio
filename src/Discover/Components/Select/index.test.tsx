import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Select from "./index";
import { countries } from "../../../tests/__mocks__/data";

const props = {
  getCountry: jest.fn(),
  country: "country-1",
  countries: countries(5),
};

describe("<Select />", () => {
  it("should render the component without crashing", () => {
    expect(shallow(<Select {...props} />));
  });
});

describe("select component data received", () => {
  const customProps = {
    ...props,
    countries: countries(1),
    getCountry: (e: any) => {
      customProps.country = e.target.name;
    },
  };
  const wrapper = mount(<Select {...customProps} />);
  it("correct data received", () => {
    expect(wrapper.find(".search__select-data").instance().value).toBe(
      customProps.countries[0].name
    );
  });
  it("selected value should be empty before click", () => {
    expect(wrapper.find("select").instance().value).toBe("");
  });
  it("selected value should not be empty after selection", () => {
    wrapper
      .find("select")
      .simulate("change", { target: customProps.countries[0] });
    expect(customProps.country).toBe(customProps.countries[0].name);
    expect(props.country).not.toBe(customProps.countries[0].name);
  });
});

describe("snapshots", () => {
  const tree = toJson(shallow(<Select {...props} />));
  expect(tree).toMatchSnapshot();
});
