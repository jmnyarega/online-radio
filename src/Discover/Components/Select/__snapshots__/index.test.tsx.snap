// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<span
  className="search-custom__select row"
>
  <select
    className="search__select-country block"
    onChange={[MockFunction]}
    value="country-1"
  >
    <option
      value=""
    >
      ---Select Country---
    </option>
    <option
      className="search__select-data"
      key="0"
      value="country-0"
    >
      country-0
    </option>
    <option
      className="search__select-data"
      key="1"
      value="country-1"
    >
      country-1
    </option>
    <option
      className="search__select-data"
      key="2"
      value="country-2"
    >
      country-2
    </option>
    <option
      className="search__select-data"
      key="3"
      value="country-3"
    >
      country-3
    </option>
    <option
      className="search__select-data"
      key="4"
      value="country-4"
    >
      country-4
    </option>
  </select>
  <span
    className="search-custom__arrow arrow"
  />
</span>
`;
